### ISSUE TITLE
Cannot detect/specify kernel in WinPython environment

### ISSUE THREAD

**Issue by 4piu** [microsoft/vscode-jupyter#15593]
### Applies To

- [X] Notebooks (.ipynb files)
- [ ] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

I am using [WinPython](https://github.com/winpython/winpython) 3.12.2 environment on my Windows 11 machine, and launch VS Code through their provided shortcut. 
I have no trouble setting up Python interpreter by manually specifying the python executable path:
![image](https://github.com/microsoft/vscode-jupyter/assets/16668056/470c8a7d-7d80-4766-af6e-4399793ce7ba)

However, notebook plugin cannot detect python environment (empty list), nor allow manually specifying path:
![image](https://github.com/microsoft/vscode-jupyter/assets/16668056/5b7ef776-7f2d-43f0-abbe-26c15ea016e1)
![image](https://github.com/microsoft/vscode-jupyter/assets/16668056/c89ec4bf-e36c-45bd-82dd-d200ae9be4d3)

### VS Code Version

1.88.1

### Jupyter Extension Version

v2024.4.2024041101

### Coding Language and Runtime Version

WinPython 3.12.2

### Running Jupyter locally or remotely?

Local

**Comment by DonJayamanne** [comment]
Does python show up in the list of Python environments when using the command `Python: Select Interpreter`?

**Comment by 4piu** [comment]
> Does python show up in the list of Python environments when using the command `Python: Select Interpreter`?

No, the `Python:Select Interpreter` only shows the `Python is not installed` warning, and I have to use `Enter interpreter path...` to specify the interpreter. 
Even after entering the path, the VS Code does not create an entry for it. (As the first screenshoot shows)

**Comment by 4piu** [comment]
I also tried setting the `python.defaultInterpreterPath`, which adds an entry to `Python: Select Interpreter`, but the vscode-jupyter's `Select Kernel > Python Environments...` list is still empty. 
![image](https://github.com/microsoft/vscode-jupyter/assets/16668056/6cd345f4-8e27-41ef-9cff-7fe4237b7f62)
